src/Update/RecipePatcher.php

```diff
@@ -39,11 +42,30 @@ public function __construct(string $rootDir, IOInterface $io)
     public function applyPatch(RecipePatch $patch): bool
     {
         $withConflicts = $this->_applyPatchFile($patch);
+        $lockedFiles = array_count_values(array_merge(...array_column($this->symfonyLock->all(), 'files')));
 
+        $nonRemovableFiles = [];
         foreach ($patch->getDeletedFiles() as $deletedFile) {
-            if (file_exists($this->rootDir.'/'.$deletedFile)) {
-                $this->execute(\sprintf('git rm %s', ProcessExecutor::escape($deletedFile)), $this->rootDir);
+            if (!file_exists($this->rootDir.'/'.$deletedFile)) {
+                continue;
+            }
+
+            if (isset($lockedFiles[$deletedFile])) {
```

Comment:
Won't this be always true, since the file will be listed all the time because the currently updated recipe has it?